scripts/Engineering/EnginX.py

```diff
@@ -175,125 +220,268 @@ def create_calibration(ceria_run, van_run, calibration_directory, calibration_ge
+    :param sample_ws: The workspace with the sample data.
+    :param vanadium_workspace: The workspace with the vanadium data
+    :param van_integration: The integration values from the vanadium corrections
+    :param bank: The bank to crop to, both if none.
+    :param spectrum_numbers: The spectrum numbers to crop to, no crop if none.
+    :param full_inst_calib : workspace containing the full instrument calibration
+    :return: The calibration output files, the vanadium curves workspace(s), and a clone of the sample file
+    """
+
+    def run_pd_calibration(kwargs_to_pass):
+        return simple.PDCalibration(**kwargs_to_pass)
+
+    def focus_and_make_van_curves(ceria_d, vanadium_d, grouping_kwarg):
+        # focus ceria
+        focused_ceria = simple.DiffractionFocussing(InputWorkspace=ceria_d, **grouping_kwarg)
+        simple.ApplyDiffCal(InstrumentWorkspace=focused_ceria, ClearCalibration=True)
+        tof_focused = simple.ConvertUnits(InputWorkspace=focused_ceria, Target='TOF')
+
+        # focus van data
+        focused_van = simple.DiffractionFocussing(InputWorkspace=vanadium_d, **grouping_kwarg)
+
+        background_van = simple.EnggEstimateFocussedBackground(InputWorkspace=focused_van, NIterations='15',
+                                                               XWindow=0.03)
+
+        simple.DeleteWorkspace(focused_ceria)
+        simple.DeleteWorkspace(focused_van)
+
+        return tof_focused, background_van
+
+    # need to clone the data as PDCalibration rebins
+    sample_raw = simple.CloneWorkspace(InputWorkspace=sample_ws)
+
+    ws_van = simple.CloneWorkspace(vanadium_workspace)
+    simple.NormaliseByCurrent(InputWorkspace=ws_van, OutputWorkspace=ws_van)
+    simple.ApplyDiffCal(InstrumentWorkspace=ws_van, CalibrationWorkspace=full_inst_calib)
+    ws_van_d = simple.ConvertUnits(InputWorkspace=ws_van, Target='dSpacing')
+
+    ws_van_d /= van_integration
+
+    # sensitivity correction for sample
+    simple.NormaliseByCurrent(InputWorkspace=sample_ws, OutputWorkspace=sample_ws)
+    simple.ApplyDiffCal(InstrumentWorkspace=sample_ws, CalibrationWorkspace=full_inst_calib)
+    ws_d = simple.ConvertUnits(InputWorkspace=sample_ws, Target='dSpacing')
+
+    ws_d /= van_integration
+
+    simple.DeleteWorkspace(van_integration)
+    simple.DeleteWorkspace(sample_ws)
+
+    kwargs = {
+        "PeakPositions": Utils.default_ceria_expected_peaks(final=True),
+        "TofBinning": [15500, -0.0003, 52000],  # using a finer binning now have better stats
+        "PeakWindow": 0.04,
+        "MinimumPeakHeight": 0.5,
+        "PeakFunction": 'BackToBackExponential',
+        "CalibrationParameters": 'DIFC+TZERO+DIFA',
+        "UseChiSq": True
+    }
+    cal_output = dict()
+    curves_output = list()
+
+    if spectrum_numbers is None:
+        if bank == '1' or bank is None:
+            df_kwarg = {"GroupingFileName": NORTH_BANK_CAL}
+            ws_d_clone = simple.CloneWorkspace(ws_d)
```

Comment:
I might be missing something but I think we don't need to clone these workspaces before focussing?